nixos/modules/services/networking/url-bot-rs.nix

```diff
@@ -28,6 +28,31 @@ in
         Configuration for url-bot-rs, see <link xlink:href="https://github.com/nuxeh/url-bot-rs#configuration-file-options"/>
         for supported values.
       '';
+      example = {
+        "foonet" = {
+          "network" = {
+            "name" = "foonet";
+            "enable" = true;
+          };
+          "connection" = {
+            "nickname" = "urlify";
+            "server" = "irc.example.com";
+            "port" = 6698;
+            "use_ssl" = true;
+          };
+        };
+      };
+    };
+
+    basicSettings = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to use basic configuration options. Can be used to connect to a
+        single IRC network, using the basic connection options provided in this
+        module. If disabled, the full configuration can be provided via the
+        settings option, for one or more IRC networks.
+      '';
```

Comment:
I don't think we should add such an option. Instead we need to rework the settings attribute to a freeformType and then we can set the defaults there. L103+ are an antipattern IMO.

Reply:
Ah, makes a lot more sense! I'll have a look shortly